### ISSUE TITLE
Some repositories are queried over http (port 80)

### ISSUE THREAD

**Issue by Profpatsch** [syl20bnr/spacemacs#2563]
This is a security issue because it can be automatically exploited and the downloaded files replaced with something potentially harmful.
Even worse since each package executes arbitrary elisp on compilation, and I wouldn’t be surprised if that has access to the whole user home folder.

**Comment by louy2** [comment]
AFAIK Spacemacs uses HTTPS to access ELPA but HTTP to access MELPA. MELPA does not support HTTPS yet, but it's in discussion: https://github.com/milkypostman/melpa/issues/2109 . Any other repos that we have to watch out?

**Comment by Profpatsch** [comment]
e.g. for the scala layer:

```
Contacting host: melpa.org:80 [2 times]
Contacting host: orgmode.org:80
Contacting host: tromey.com:80
```

**Comment by StreakyCobra** [comment]
This has been fixed in develop by #3749

**Comment by louy2** [comment]
Shall we keep somewhere a list of the repos we use and their HTTPS support status?
